### PR TITLE
feature/SOF 7393 1

### PR DESCRIPTION
**Pull request by VsevolodX**
- feat: add island builder
- update: add island config
- update: add slab defect enum enrty

### CODE REVIEW

File: wheel_server.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import os
```
Comment (timurbazhirov) [inline wheel_server.py:1]: Move to mat3ra-utils and make it callable [project.scripts]

File: src/py/mat3ra/made/tools/build/defect/builders.py [hunk 2]
```diff
@@ -308,3 +308,46 @@ def create_adatom(
...
+        new_material = material.clone()
+        new_material.to_cartesian()
+        original_max_z = get_atomic_coordinates_extremum(new_material, use_cartesian_coordinates=True)
+        material_with_additional_layers = self.create_material_with_additional_layers(new_material, thickness)
+        added_layers_max_z = get_atomic_coordinates_extremum(material_with_additional_layers)
+
+        # Get the atoms within the sphere
+        center_coordinate = position_on_surface.copy()
+        center_coordinate.append(original_max_z)
+        atoms_within_sphere = filter_by_sphere(
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/builders.py:339]: This can be more general through any filter

File: pyproject.toml [hunk 3]
```diff
@@ -55,6 +55,7 @@ all = [
 # Entrypoint scripts can be defined here, see examples below.
 [project.scripts]
 # my-script = "my_package.my_module:my_function"
+wheel_server = "mat3ra.utils.wheel_server:main"
```
Comment (timurbazhirov) [inline pyproject.toml:58]: This should be in utils

File: src/py/mat3ra/made/tools/build/slab/configuration.py [hunk 4]
```diff
@@ -39,11 +39,13 @@ def __init__(
...
         thickness=thickness,
         vacuum=vacuum,
-        xy_supercell_matrix=xy_supercell_matrix,
+        xy_supercell_matrix=None,
         use_conventional_cell=use_conventional_cell,
         use_orthogonal_z=use_orthogonal_z,
         make_primitive=make_primitive,
     ):
+        if xy_supercell_matrix is None:
+            xy_supercell_matrix = [[1, 0], [0, 1]]
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/slab/configuration.py:48]: Use np eye https://numpy.org/doc/stable/reference/generated/numpy.eye.html

File: src/py/mat3ra/made/tools/build/defect/configuration.py [hunk 5]
```diff
@@ -93,3 +101,20 @@ def _json(self):
             **super()._json,
             "type": "AdatomSlabPointDefectConfiguration",
         }
+
+
+class IslandSlabDefectConfiguration(SlabDefectConfiguration):
+    defect_type: SlabDefectTypeEnum = SlabDefectTypeEnum.ISLAND
+    condition: Optional[Tuple[Callable[[List[float]], bool], Dict]] = CoordinateConditionBuilder().cylinder()
+    thickness: int = 1
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/configuration.py:109]: What is thickness in, which units??

Comment (VsevolodX) [reply src/py/mat3ra/made/tools/build/defect/configuration.py:109]: Layers

Comment (VsevolodX) [reply src/py/mat3ra/made/tools/build/defect/configuration.py:109]: Docs have that

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**